### DISCUSSION TITLE
How do you separate Mint code in a plurality of files (or how do you "import" code between Mint files) ?

### DISCUSSION THREAD

**Discussion by serge-hulne** [mint-lang/mint#561] [Answered] [answered]
How do you separate Mint code in a plurality of files (or how do you "import" code between Mint files) ?

For instance, how would you put the module Greet in a separate file and call it in the main module file

```mint
component Main {
  fun render : Html {
    <div>
      <button onClick={Greet.hello}>
        "Click here"
      </button>
    </div>
  }
}

module Greet {
  fun hello(event : Html.Event) {
    Debug.log("Hello")
  }
}
```

**Comment by gdotdesign** [comment, marked as answer]
> How do you separate Mint code in a plurality of files (or how do you "import" code between Mint files) ?

Every Mint project has a `mint.json` file which specifies the directories which contain source code: https://github.com/mint-lang/mint-realworld/blob/master/mint.json#L4

All `*.mint` files are parsed in the source directories and all entities are available in all files, so there is no need to import. 

> For instance, how would you put the module Greet in a separate file and call it in the main module file

Just create a new file `Greet.mint` (the name doesn't matter) in the `source` directory (which is the default) put the module there and that's it.
